packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -620,7 +623,11 @@ export function patchFetch({
           ) {
             const forceDynamic = staticGenerationStore.forceDynamic
 
-            if (!forceDynamic && next.revalidate === 0) {
+            if (
+              !forceDynamic &&
```

Comment:
nit: can probably get rid of this variable and just reference `staticGenerationStore.forceDynamic`, makes it a bit more obvious too that both values are coming from the static generation context